cmake/Modules/FindDPCPP.cmake

```diff
@@ -66,6 +67,10 @@ else()
     INTERFACE_INCLUDE_DIRECTORIES "${DPCPP_BIN_DIR}/../include/sycl")
 endif()
 
+if (DPCPP_SYCL_TARGET MATCHES "nvptx64-nvidia-cuda")
```

Comment:
Hi @muhammad-tanvir-1211, read this first.

I suggest to
* check also if `DPCPP_SYCL_ARCH >= sm_80`, see the comment below;
* specify in the README that the user need to set `DPCPP_SYCL_ARCH >= sm_80` in order to use joint matrices. If the user doesn't specify `DPCPP_SYCL_ARCH` this defaults to `sm_50`.

Reply:
Addressed these changes in the latest commits.